facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java

```diff
@@ -316,22 +324,83 @@ public void clear(int id) {
         milestone.getDeliveredArtifacts().clear();
     }
 
-    private org.jboss.pnc.model.Artifact findOrCreateNotFoundArtifact(Artifact artifact, TargetRepository targetRepo) {
-        return findOrCreateArtifact(mapNotFoundArtifact(artifact), targetRepo);
+    private Optional<org.jboss.pnc.model.Artifact> getBestMatchingArtifact(
+            Collection<org.jboss.pnc.model.Artifact> artifacts) {
+        if (artifacts == null || artifacts.isEmpty()) {
+            return Optional.empty();
+        }
+
+        if (artifacts.size() == 1) {
+            return artifacts.stream().findFirst();
+        }
+
+        return artifacts.stream()
+                .sorted(Comparator.comparing(DeliverableAnalyzerManagerImpl::getNotFoundArtifactRating).reversed())
+                .findFirst();
```

Comment:
Looking at this, the special case for size 1 is not technically necessary. But maybe it's more understandable like this?